text/0000-terminology-and-boundaries.md

```diff
@@ -0,0 +1,224 @@
+- Feature Name: Terminology and Boundary Definition
+- Start Date: 2018-10-11
+- RFC PR: (leave this empty)
+- Nebulet Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+The Nebulet operating system, being so different from popular operating systems
+of today, will require new terminology to refer to abstraction boundaries.
+Terms such as "userspace" and "kernelspace" will carry too much baggage from
+existing OSes, which is likely to confuse potential contributors. This RFC
+proposes we define the following terms to refer to parts of our OS, adapted
+from [EROS](https://web.archive.org/web/20160412201504/http://www.coyotos.org/docs/misc/eros-structure.pdf):
+the _kernel_, which implements threading and inter-proces communication (IPC)
+as a wasm module, the _nucleus_, which implements a small wasm JIT natively,
+and the _firmament_, which specifies IPC interfaces for drivers to implement.
+Applications will be implemented atop these abstractions.
+
+# Motivation
+[motivation]: #motivation
+
+Today's common operating systems generally use hardware-based isolation
+approaches, such as protection rings on x86 or processor modes on ARM, to
+protect the kernel's memory from tampering. In colloquial operating systems
+discussion, the phrases "kernelspace" and "ring 0" have become synonymous,
+as have the phrases "userspace" and "ring 3". As Nebulet uses software-based
+isolation and not protection rings, these terms will be unfortunately
+overloaded, confusing potential contributors. The world of microkernel-based
+research OSes has provided some alternative terminology, some of which is
+potentially useful here.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The Nebulet operating system architecture is defined in terms of four
+components:
+
+- The _kernel_, which implements the core functionality of a microkernel OS:
```

Comment:
Can you be clearer about whether the kernel is a wasm module or not?

Reply:
Done in latest commit.